depscan/cli.py

```diff
@@ -581,78 +584,119 @@ async def run_scan():
-    if not project_type and params.get("type"):
-        project_type = params.get("type")
-    if not profile and params.get("profile"):
-        profile = params.get("profile")
-    if not path and not url:
-        return {"error": "true", "message": "path or url is required"}, 500
+    if params is not None:
+        if not url and params.get("url"):
+            url = params.get("url")
+        if not path and params.get("path"):
+            path = params.get("path")
+        if not multi_project and params.get("multiProject"):
+            multi_project = params.get("multiProject", "").lower() in ("true", "1")
+        if not project_type and params.get("type"):
+            project_type = params.get("type")
+        if not profile and params.get("profile"):
+            profile = params.get("profile")
+
+    if not path and not url and (uploaded_bom_file.get('file', None) is None):
+        return {"error": "true", "message": "path or url or a bom file upload is required"}, 400
+    if not project_type:
+        return {"error": "true", "message": "project type is required"}, 400
+
     if not db_lib.index_count(db["index_file"]):
         return {
             "error": "true",
             "message": "Vulnerability database is empty. Prepare the "
             "vulnerability database by invoking /cache endpoint "
             "before running scans.",
-        }, 500
+        }, 500, {"Content-Type": "application/json"}
+
     cdxgen_server = app.config.get("CDXGEN_SERVER_URL")
-    with tempfile.NamedTemporaryFile(delete=False, suffix=".bom.json") as bfp:
-        bom_status = create_bom(
-            project_type,
-            bfp.name,
-            path,
-            True,
-            {
-                "url": url,
-                "path": path,
-                "type": project_type,
-                "multiProject": multi_project,
-                "cdxgen_server": cdxgen_server,
-                "profile": profile,
-            },
-        )
-        if bom_status:
-            LOG.debug("BOM file was generated successfully at %s", bfp.name)
-            pkg_list = get_pkg_list(bfp.name)
-            if not pkg_list:
-                return {}
-            if project_type in type_audit_map:
-                audit_results = audit(project_type, pkg_list)
-                if audit_results:
-                    results = results + audit_results
-            vdb_results, pkg_aliases, sug_version_dict, purl_aliases = scan(
-                db, project_type, pkg_list, True
-            )
-            if vdb_results:
-                results += vdb_results
-            results = [r.to_dict() for r in results]
-            bom_data = json.load(bfp)
-            options = PrepareVdrOptions(
+    bom_file_path = None
+
+    if uploaded_bom_file.get('file', None) is not None:
+        bom_file = uploaded_bom_file['file']
+        bom_file_content = bom_file.read().decode('utf-8')
```

Comment:
At this point the uploaded file could be of any extension including videos and binaries. Is there another eay to configure quart to only accept specific mimetypes?